### PR TITLE
feat: add deribit options api - closes #25

### LINKED ISSUE #25: Add Deribit Options API

### PR DESCRIPTION
**Pull request by IlluvatarEru**
resolves #25

### CODE REVIEW

File: rest/test/test_deribit_option.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        self.assertGreater(tob_ask, tob_bid)
+
+        tob_mid = deribit_option_api.get_tob_mid(INSTRUMENT)
+        self.assertGreater(tob_mid, tob_bid)
+        self.assertLess(tob_mid, tob_ask)
+
+        tob_spread = deribit_option_api.get_tob_spread(INSTRUMENT)
+        self.assertGreater(tob_spread, 0)
+
+    def test_orderbook(self):
```
Comment (yvesleconte) [inline rest/test/test_deribit_option.py:51]: Test fails:     
self.assertEqual(len(ob[ASK_SIZES].values[0]), n_levels)
AssertionError: 0 != 1

File: rest/src/request_types.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
 POST = 'POST'
 
 
+def dict_to_querystring(params):
+    return '&'.join([f'{k}={v}' for k, v in params.items()])
```
Comment (yvesleconte) [inline rest/src/request_types.py:6]: not very understandable variable names

**Review by yvesleconte — CHANGES_REQUESTED** [review]

File: rest/src/request_types.py (hunk 2 above)
Comment (IlluvatarEru) [reply rest/src/request_types.py:6]: actually it's quite common for short functions involving dictionaries to use `k` for key and `v` for value

File: rest/test/test_deribit_option.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        self.assertEqual(str(context.exception).split(":")[0],
+                         "Option X_ETHUSD_230630_1200 has wrong format, it should be C_LHSRHS_YYMMDD_K or P_LHSRHS_YYMMDD_K")
+
+        wrong_sym = "C_XXXUSD_230630_1200"
+        with self.assertRaises(Exception) as context:
+            tob_bid = deribit_option_api.get_tob_bid(wrong_sym)
+        self.assertEqual(str(context.exception).split(":")[0],
+                         "Option C_XXXUSD_230630_1200 has wrong format, currency pair XXXUSD not supported")
+
+    def test_tob_prices(self):
```
[the remark this answers is not in the document]
Comment (IlluvatarEru) [reply rest/test/test_deribit_option.py:36]: Needs to give more details that "Test fails", a screenshot or a copy paste of the trace. It works fine for me.

File: core/src/option_syms.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+def check_currency_pair_option(sym):
+    pattern = re.compile("_")
+    if len(pattern.findall(sym)) != 3:
+        raise_wrong_format_option(sym)
+    if sym[:2] not in [C_, P_]:
+        raise_wrong_format_option(sym)
+    spot = sym.split("_")[1]
+    if spot not in SUPPORTED_CCY_PAIRS:
+        print(spot)
```
[the remark this answers is not in the document]
Comment (IlluvatarEru) [reply core/src/option_syms.py:28]: fixed

**Comment by yvesleconte** [comment]
Here are some errors I recently encountered with the unittests:

Testing started at 23:13 ...
Launching trial with arguments --reporter=teamcity test_deribit_option.TestDeribitOption in C:\PATH\barad-dur\rest\test


url=https://deribit.com/api/v2/public/get_tradingview_chart_data?instrument_name=ETH-30JUN23-1200-C&start_timestamp=1680297237&end_timestamp=1680297237578&resolution=1D

params={'instrument_name': 'ETH-30JUN23-1200-C', 'start_timestamp': 1680297237, 'end_timestamp': 1680297237578, 'resolution': '1D'}

url=https://deribit.com/api/v2/public/get_tradingview_chart_data?instrument_name=ETH-30JUN23-1200-C&start_timestamp=1680297237&end_timestamp=1680297237833&resolution=1D

params={'instrument_name': 'ETH-30JUN23-1200-C', 'start_timestamp': 1680297237, 'end_timestamp': 1680297237833, 'resolution': '1D'}

Failure
Traceback (most recent call last):
  File "C:\PATH\barad-dur\rest\test\test_deribit_option.py", line 110, in test_close
    self.assertEqual(close_price, 0.32)
AssertionError: 0.364 != 0.32


url=https://deribit.com/api/v2/public/get_tradingview_chart_data?instrument_name=ETH-30JUN23-1200-C&start_timestamp=1680040800&end_timestamp=1680297237921&resolution=1D

params={'instrument_name': 'ETH-30JUN23-1200-C', 'start_timestamp': 1680040800, 'end_timestamp': 1680297237921, 'resolution': '1D'}
start_date=2022-12-21

url=https://deribit.com/api/v2/public/get_tradingview_chart_data?instrument_name=ETH-30JUN23-1200-C&start_timestamp=1671663600000&end_timestamp=1680297238139&resolution=1D

params={'instrument_name': 'ETH-30JUN23-1200-C', 'start_timestamp': 1671663600000, 'end_timestamp': 1680297238139, 'resolution': '1D'}

url=https://deribit.com/api/v2/public/get_order_book?instrument_name=ETH-30JUN23-1200-C&depth=1

params={'instrument_name': 'ETH-30JUN23-1200-C', 'depth': 1}

Failure
Traceback (most recent call last):
  File "C:\PATH\barad-dur\rest\test\test_deribit_option.py", line 62, in test_orderbook
    self.assertEqual(len(ob[ASK_SIZES].values[0]), n_levels)
AssertionError: 0 != 1


url=https://deribit.com/api/v2/public/get_order_book?instrument_name=ETH-30JUN23-1200-C&depth=1

params={'instrument_name': 'ETH-30JUN23-1200-C', 'depth': 1}

url=https://deribit.com/api/v2/public/get_order_book?instrument_name=ETH-30JUN23-1200-C&depth=1

params={'instrument_name': 'ETH-30JUN23-1200-C', 'depth': 1}

Error
Traceback (most recent call last):
Failure: builtins.tuple: (<class 'IndexError'>, IndexError('list index out of range'), <traceback object at 0x000002848EB87348>)

XXXUSD





===============================================================================
[FAIL]
Traceback (most recent call last):
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 59, in testPartExecutor
    yield
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 628, in run
    testMethod()
  File "C:\PATH\barad-dur\rest\test\test_deribit_option.py", line 110, in test_close
    self.assertEqual(close_price, 0.32)
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)
builtins.AssertionError: 0.364 != 0.32

test_deribit_option.TestDeribitOption.test_close
===============================================================================
[FAIL]
Traceback (most recent call last):
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 59, in testPartExecutor
    yield
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 628, in run
    testMethod()
  File "C:\PATH\barad-dur\rest\test\test_deribit_option.py", line 62, in test_orderbook
    self.assertEqual(len(ob[ASK_SIZES].values[0]), n_levels)
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 852, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\Users\Yves\AppData\Local\Programs\Python\Python37\lib\unittest\case.py", line 845, in _baseAssertEqual
    raise self.failureException(msg)
builtins.AssertionError: 0 != 1

test_deribit_option.TestDeribitOption.test_orderbook
===============================================================================
[ERROR]
Traceback (most recent call last):
Failure: builtins.tuple: (<class 'IndexError'>, IndexError('list index out of range'), <traceback object at 0x000002848EB87348>)

test_deribit_option.TestDeribitOption.test_tob_prices
-------------------------------------------------------------------------------
Ran 6 tests in 1.277s

FAILED (failures=2, errors=1, successes=3)

Process finished with exit code 1

**Pull request merged**